ArduCopter/gps_glitch.pde

```diff
@@ -17,6 +17,14 @@ static void gps_glitch_update() {
     }
 }
 
+static void gps_glitch_mode_change_commanded(uint8_t mode_commanded)
+{
+    // ensure that commanded mode switches to ALT_HOLD will cancel the switch back to FLY
+    if (mode_commanded == ALT_HOLD) {
```

Comment:
Yeah, I could eliminate that check in the update function. I'd want to disable the switch only on successful changes, in case e.g. shots manager sends a guided mode request or similar.

I could change it to something like the below, but I think this PR is fine as-is...

    gps_glitch_mode_change_commanded(uint8_t mode_commanded, uint8_t success) {
        if (mode_commanded == ALT_HOLD) {
            gps_glitch_switch_mode_on_resolve = false;
        } else if (success && mode_commanded != LOITER) {
            gps_glitch_switch_mode_on_resolve = false;
        }
    }